common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -1268,19 +1273,21 @@ def _update_single_item(self, location, update, allow_not_found=False):
         Set update on the specified item, and raises ItemNotFoundError
         if the location doesn't exist
         """
-        bulk_record = self._get_bulk_ops_record(location.course_key)
-        bulk_record.dirty = True
-        # See http://www.mongodb.org/display/DOCS/Updating for
-        # atomic update syntax
-        result = self.collection.update(
-            {'_id': location.to_deprecated_son()},
-            {'$set': update},
-            multi=False,
-            upsert=allow_not_found,
-            w=1,  # wait until primary commits
-        )
-        if result['n'] == 0:
-            raise ItemNotFoundError(location)
+        course_key = location.course_key
+        with self.bulk_operations(course_key):
```

Comment:
I believe the addition of this line here will cause the Publish signal to fire whenever a change is made in an old Mongo course - not just when a Published version of a module is changed.  So as course authors work on the draft version of a course, the signal will fire for each change.  Have we tested this?